File: sort/radix/py/radix_sort.py
import sort.count.py.counting_sort as csort

# digits means the number of digits for ints in A.
def sort(A, digits):
    for i in range(digits):
        B = csort.sort(A, 9, lambda a: (a / 10 ** i) % 10)
        A = B
    return A

File: sort/radix/java/RadixSort.java
public class RadixSort {
  public static int[] sort(int[] arr, int digits) {
    for (int i = 0; i < digits; i++) {
      MapInt mapper = new DigitMapper(i);
      arr = CountingSort.sort(arr, 9, mapper);
    }
    return arr;
  }
}

class DigitMapper implements MapInt {
  public DigitMapper(int digit) {
    this.digit = digit;
  }

  public int map(int e) {
    int d = this.digit;
    while (d-- > 0) {
      e /= 10;
    }
    return e % 10;
  }

  private int digit;
}
